pkg/webhook/resources/virtualmachinerestore/validator.go

```diff
@@ -169,6 +172,27 @@ func (v *restoreValidator) handleNewVM(newRestore *v1beta1.VirtualMachineRestore
 	if err := v.vmrCalculator.CheckIfVMCanStartByResourceQuota(vm); err != nil {
 		return werror.NewInternalError(fmt.Sprintf("Failed to restore the new vm, err: %+v", err))
 	}
+
+	// we don't have a global macaddress checker,
+	// so we just check the source vm name and vmrestores which using the same vmbackup.
+	// this can be removed after https://github.com/harvester/harvester/issues/4893
+	otherVMRestores, err := v.vmRestore.GetByIndex(indexeres.VMRestoreByVMBackupNamespaceAndName, fmt.Sprintf("%s-%s", vmRestore.Spec.VirtualMachineBackupNamespace, vmRestore.Spec.VirtualMachineBackupName))
+	if err != nil {
+		return werror.NewInternalError(fmt.Sprintf("failed to get vmrestore by index %s, err: %+v", indexeres.VMRestoreByVMBackupNamespaceAndName, err))
+	}
+	for _, otherVMRestore := range otherVMRestores {
+		if otherVMRestore.Spec.NewVM && otherVMRestore.Spec.KeepMacAddress {
+			return werror.NewInvalidError(fmt.Sprintf("can't restore the new vm with the same macaddress as the vmrestore %s/%s", otherVMRestore.Namespace, otherVMRestore.Name), fieldKeepMacAddress)
+		}
+	}
+	sourceVM, err := v.vms.Get(vmBackup.Namespace, vmBackup.Spec.Source.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return werror.NewInternalError(fmt.Sprintf("failed to get the VM %s/%s, err: %+v", vmBackup.Namespace, vmBackup.Spec.Source.Name, err))
+	}
+	if sourceVM != nil && vmBackup.Status.SourceUID != nil && sourceVM.UID == *vmBackup.Status.SourceUID {
+		return werror.NewInvalidError("can't restore the new vm with the same macaddress because the source vm is still existent", fieldKeepMacAddress)
+	}
+
 	return nil
```

Comment:
Just a suggestion for readability:
```
	sourceVM, err := v.vms.Get(vmBackup.Namespace, vmBackup.Spec.Source.Name)
	if apierrors.IsNotFound(err) {
		return nil
	}
	if err != nil {
		return werror.NewInternalError(fmt.Sprintf("failed to get the VM %s/%s, err: %+v", vmBackup.Namespace, vmBackup.Spec.Source.Name, err))
	}
	if sourceVM != nil && vmBackup.Status.SourceUID != nil && sourceVM.UID == *vmBackup.Status.SourceUID {
		return werror.NewInvalidError("can't restore the new vm with the same macaddress because the source vm is still existent", fieldKeepMacAddress)
	}
```

Reply:
Updated it. Thank you.